### PR TITLE
Feature/heartbeat

### PR DESCRIPTION
**Pull request by cristipogacean**

### CODE REVIEW

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Extensions/PublishedDataItemsModelEx.cs [hunk 1]
```diff
@@ -31,9 +31,9 @@ public static class PublishedDataItemsModelEx {
                     continue;
                 }
                 var monitoredItem = item.ToMonitoredItem();
-                map.Add(monitoredItem.Id ?? Guid.NewGuid().ToString(),
-                    monitoredItem);
-                if (item.HeartbeatInterval == null) {
+                map.Add(monitoredItem.Id ?? Guid.NewGuid().ToString(), monitoredItem);
+                /*  Heartbeat using triggering mode
```
Comment (marcschier) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Extensions/PublishedDataItemsModelEx.cs:35]: If we never believe this will be using triggering lets just remove the commented out code.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Extensions/MonitoredItemNotificationModelEx.cs [hunk 2]
```diff
@@ -35,10 +35,6 @@ public static class MonitoredItemNotificationModelEx {
                 if (message == null) {
                     continue;
                 }
-                if (message.DiagnosticInfo == null && notification.DiagnosticInfos != null &&
```
Comment (marcschier) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Extensions/MonitoredItemNotificationModelEx.cs:38]: Why did you remove this, was it causing issues or is it never null?   We want to flow the diagnostics info back to the configuration eventually so having this does not hurt on the way there.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Extensions/PublishedDataItemsModelEx.cs (hunk 1 above)
Comment (cristipogacean) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Extensions/PublishedDataItemsModelEx.cs:35]: let's keep it as sample for generic triggering feature, I think it makes a lot of sense to promote this feature.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Extensions/MonitoredItemNotificationModelEx.cs (hunk 2 above)
Comment (cristipogacean) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Extensions/MonitoredItemNotificationModelEx.cs:38]: The diagnostics (if any) are already added to the message through: public static MonitoredItemNotificationModel ToMonitoredItemNotification(). Diagnostics are copied from the DataChangeNotification.DiagnosticInfo array to the MonitoredItemNotification.DiagnosticInfo by the stack before the callback is triggered.
